src/main/java/com/spotify/folsom/client/Request.java

```diff
@@ -65,4 +72,32 @@ protected static ByteBuf toBuffer(final ByteBufAllocator alloc, ByteBuffer dst,
   public int getOpaque() {
     return opaque;
   }
+
+  @VisibleForTesting
+  static byte[] encodeKey(String key, Charset charset) {
+    checkNotNull(key, "key");
+    byte[] keyBytes = key.getBytes(charset);
+    int length = keyBytes.length;
+    if (length > 250) {
+      throw new IllegalArgumentException("Key is too long: " + key);
+    }
+    if (length <= 0) {
+      throw new IllegalArgumentException("Key is empty");
+    }
+    for (int i = 0; i < length; i++) {
+      final byte c = keyBytes[i];
+      if (c >= 0 && c <= 32) {
+        throw new IllegalArgumentException("Invalid key: " + key);
```

Comment:
Good question! I am fairly sure memcached treats the input as a stream of 8-bit characters, and only explicitly disallowing space and some control values.

This probably implies that UTF-16 won't work for memcached, assuming some valid utf-16 character is represented as 0x??20 (where ?? can be anything)